File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String , List<String>> map = new HashMap<>();
        for(int i = 0 ; i < strs.length ; i++){
            int[] alph = new int[26];
            for(int j = 0 ; j < strs[i].length() ;j++){
                alph[strs[i].charAt(j) - 'a']++;
            }
            String x = Arrays.toString(alph);
            if(!map.containsKey(x)) map.put(x , new ArrayList<>());
            map.get(x).add(strs[i]);
        }
        List<List<String>> sol = new ArrayList<>();
        for(List<String> l : map.values()){
            sol.add(l);
        }
        return sol;
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        s = strs[0]
        hashmap = {}
        for s in strs:
            chars = [0] * 26
            for c in s:
                chars[ord(c) - ord('a')] += 1
            arr = hashmap.setdefault(tuple(chars), [])
            arr.append(s)
        return list(hashmap.values())
        